### PR TITLE
Use CSS for the contact details popup

### PR DESCRIPTION
**Pull request by siefkenj**
Removes the complicated logic in the `Fade` component and instead uses CSS transitions to fade the contact details in and out (fade seems to have been broken for a while anyways...).

This PR also replaces the `_getSeparator` logic with `Intl.ListFormat`. I can't quite tell, but `header.commaSeparator` and `header.andSeparator` might be able to be removed after this is merged.

I also attempted to clean up some of the `ContactDetails` object. It's still confusing...can you lay out a standard of what it's supposed to look like given what properties? If I know when it's supposed to display what, I can write some tests.

### CODE REVIEW

**Comment by siefkenj** [comment]
I believe I fixed the issue with the overflowing popups. Everything should work now :-)

**Review by Standard8 — CHANGES_REQUESTED** [review]
Did you use an old branch to base this on? The split options to show extra buttons at the bottom of the contacts popup seem to have come back. I got rid of that split a while ago, as I didn't think it was worth it.

The other thing I'm not quite sure about is when it gets shown. With this version of the patch if I move my mouse vertically across the contact, then the pop-up will start showing straight away.

I tried experimenting with a 400ms delay (roughly around the time of the current delay iirc), and it was better, though I was still finding that if you moved the mouse down and stopped before you get past the contact, then the contact detail was still prone to being displayed.

I'm not quite sure yet exactly how bad that is overall. Might need some more testing time.

Also I noticed that the "createFilter" text wasn't being replaced on the button.

**Comment by siefkenj** [comment]
Yes, I did use an old version...(I was finishing up the work I started a year ago...). I'll update stuff to be in line with the new version and add a delay.

**Comment by siefkenj** [comment]
Okay, hopefully things are good now!

File: addon/content/messageHeader.jsx [hunk 1]
```diff
@@ -11,327 +12,384 @@ import { MessageTags, SpecialMessageTags } from "./messageTags.jsx";
...
+      </span>
+      {popupParentNode &&
+        ReactDOM.createPortal(
+          <div
+            className={`fade-popup ${shouldShowPopup ? "hover" : ""}`}
+            style={{
+              //left: window.scrollX + pos.left,
+              //top: window.scrollY + pos.bottom,
+              //left: spanRef.current?.offsetLeft,
+              //top: spanRef.current?.offsetTop + spanRef.current?.offsetHeight,
```
Comment (Standard8) [inline addon/content/messageHeader.jsx:102]: Can we drop these lines now?

File: addon/content/messageHeader.jsx [hunk 2]
```diff
@@ -11,327 +12,384 @@ import { MessageTags, SpecialMessageTags } from "./messageTags.jsx";
...
+        </span>
+      </span>
+    </HoverFade>
+  );
+}
+DetailedContactLabel.propTypes = {
+  className: PropTypes.string.isRequired,
+  contact: PropTypes.object.isRequired,
+};
+export function ContactLabel({ contact, className }) {
```
Comment (Standard8) [inline addon/content/messageHeader.jsx:176]: nit: Line break before the export please

**Review by Standard8 — CHANGES_REQUESTED** [review]
I've been trying this over the last week and not noticed any issues, so lets get this landed.

Just a minor couple of things to fix.

File: addon/content/messageHeader.jsx (hunk 1 above)
Comment (siefkenj) [reply addon/content/messageHeader.jsx:102]: Oops!

**Review by Standard8 — APPROVED** [review]

**Pull request merged**